framework/web/Request.php

```diff
@@ -375,7 +375,11 @@ public function getHeaders()
                     if (strncmp($name, 'HTTP_', 5) === 0) {
                         $name = str_replace(' ', '-', ucwords(strtolower(str_replace('_', ' ', substr($name, 5)))));
                         $this->_headers->add($name, $value);
+                    } elseif (strncmp($name, 'REDIRECT_HTTP_', 14) === 0) {
```

Comment:
Do we expect something else here than `REDIRECT_HTTP_AUTHORIZATION`? If not - there is no reason for the next line to be so sophisticated.

Reply:
Sounds like a prequel to the next bug. Request::getHeaders() is a global headers parser method, so we can expect any header here that were set in RewriteRule as environment variable.

Reply:
Not sure if I understand what you said but I think you mean that there can be something like `REDIRECT_HTTP_WHATEVER` and we should add `Whatever` then to headers. If so, then ok, I got it now.

Reply:
We can parse only "REDIRECT_" prefix for any headers, so the rules for Authorization header will become
```
RewriteRule .* - [E=AUTHORIZATION:%{HTTP:Authorization}]
SetEnvIf Authorization .+ AUTHORIZATION=$0
```
and env var name will become REDIRECT_AUTHORIZATION.

As I understand HTTP_AUTHORIZATION was used in the Rewrite rule later, before Apache start prefixes env vars with "REDIRECT_".

Reply:
> Not sure if I understand what you said but I think you mean that there can be something like `REDIRECT_HTTP_WHATEVER` and we should add `Whatever` then to headers. If so, then ok, I got it now.

In this rule `SetEnvIf Authorization .+ AUTHORIZATION=$0` you can set any env var name for any http header.